### PR TITLE
[Commands] Add swift package clean

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
and deprecate swift build --clean

- https://bugs.swift.org/browse/SR-2082

### CODE REVIEW

File: Sources/Commands/SwiftBuildTool.swift [hunk 1]
```diff
@@ -43,14 +43,8 @@ public class SwiftBuildTool: SwiftTool<BuildToolOptions> {
...
         case .clean:
-            if options.enableNewResolver {
-                try getActiveWorkspace().clean()
-            } else {
-                // FIXME: This test is lame, `removeFileTree` shouldn't error on this.
-                if exists(buildPath) {
-                    try removeFileTree(buildPath)
-                }
-            }
+            print("warning: swift build --clean is deprecated. Use 'swift package clean' instead.")
```
Comment (ddunbar) [inline Sources/Commands/SwiftBuildTool.swift]: Let's adopt a practice of including the SR # in the diagnostic when we deprecate things.

File: Sources/Commands/SwiftBuildTool.swift [hunk 2]
```diff
@@ -63,7 +57,7 @@ public class SwiftBuildTool: SwiftTool<BuildToolOptions> {
             to: { $0.buildTests = $1 })
 
         binder.bind(
-            option: parser.add(option: "--clean", kind: Bool.self, usage: "Delete build artifacts"),
+            option: parser.add(option: "--clean", kind: Bool.self, usage: "Delete build artifacts (Deprecated: Use 'swift package clean')"),
             to: { $0.clean = $1 })
```
Comment (ddunbar) [inline Sources/Commands/SwiftBuildTool.swift]: Can we just drop usage once it becomes deprecated?

Comment (aciidb0mb3r) [inline Sources/Commands/SwiftBuildTool.swift]: sure

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test os x

**Comment by aciidb0mb3r** [comment]
Tested osx locally

**Pull request merged**